pam_authz/README.md

```diff
@@ -1,174 +1,28 @@
-group (as defined in policy) have ssh access to any server and can perform sudo
-commands. The user `ops-dev` is an admin, so run the following commands and see
-that they all succeed.
-
-```
-# ssh into webapp, with sudo
-$ ssh -i docker/id_rsa -p 2222 ops-dev@$docker_ip
-$ sudo ls /
-$ logout
-# ssh into backend, with sudo
-$ ssh -i docker/id_rsa -p 2223 ops-dev@$docker_ip
-$ sudo ls /
-$ logout
-```
-
-Let's try a user not in the admin group. Recall that a non-admin can ssh into
-any server running an app they wrote code for.   The user `web-dev` contributes
-to the webapp server, but not the backend. `web-dev` can ssh into servers with
-the `webapp` role, but not any other servers. Further, this user should not be
-able to use sudo.
-
-```
-# ssh into webapp, but no sudo
-$ ssh -i docker/id_rsa -p 2222 web-dev@$docker_ip
-$ sudo ls /
-$ logout
-# no ssh into backend
-$ ssh -i docker/id_rsa -p 2223 web-dev@$docker_ip
-```
-
-To this point, the policy could have been hard coded into the PAM module. OPA,
-however, allows us to dynamically modify the policies and the data that policies
-rely on. Consider the user "Pam". In our initial data, Pam has contributed to
-the WebApp, so can only ssh into the container with the webapp role. Suppose
-that Pam makes changes to the backend as well, and so Pam should be added to the
-list of Backend contributers. We use OPA's API to update the data like so:
-
-```
-curl -X PUT $docker_ip:8181/v1/data/io/vcs/contributers -d \
-'{
-    "WebApp": {
-        "Contributers": ["web-dev", "Pam", "Hans"]
-    },
-    "Backend": {
-        "Contributers": ["Pam", "backend-dev", "Stan", "Hans"]
-    }
-}'
-```
-
-Now `ssh -i docker/id_rsa -p 2223 web-dev@$docker_ip` is allowed.
-
-As another example, suppose that a bug occurs in production and Pam needs sudo
-access to debug. We can update the list of admins to give Pam sudo access like
-so:
-
-```
-curl -X PUT $docker_ip:8181/v1/data/io/directory/users -d \
-'{
-    "engineering": ["web-dev", "backend-dev" ,"ops-dev", "Stan", "Hans", "Pam", "Sam", "Jan"],
-    "admin": ["Pam", "ops-dev"]
-}'
-```
-
-Now Pam has ssh and sudo access on all servers.
-
-
-### Orchestration of Multiple Servers
-
-The previous example shows how OPA and PAM can be used for authorization on a
-single server, but in practice policy needs to be enforced across many servers.
-Because OPA is a host-local daemon, we recommend using a standalone, centralized
-service (like etcd) to store policy and data centrally and a side-car (or
-wrapper) for OPA that keeps all the OPAs up to date with that central service.
-(OPA was purpose-built to be a host-local daemon to ensure high-availability and
-high-performance even in the presence of network partitions and leaves the
-problem of policy/data replication up to the environment in which it is
-deployed.)
-
-
-
+After this completes, `docker images` will show the
+following images: `ssh-backend`, `ssh-webapp`, and `pam-builder`. The builder
```

Comment:
The images are now `openpolicyagent/demo-pam` and `pam-builder`. The `demo-pam` image now covers both the webapp and backend roles (just the json file mounted into each running container is different using docker compose).